File: source/gui/helper/GUIDisplay.java
package gui.helper;

import glparts.DepthTexture;
import glparts.Texture;
import imgui.ImGui;
import math.Float2;
import math.Int2;

public final class GUIDisplay {
    private GUIDisplay() {}

    public static void texture(Texture texture, Int2 size) {
        GUIDisplay.texture(texture, new Float2(size));
    }

    public static void texture(DepthTexture texture, Int2 size) {
        GUIDisplay.texture(texture, new Float2(size));
    }

    public static void texture(Texture texture, Float2 size) {
        GUIDisplay.texture(texture.getBuffer(), size.x, size.y);
    }

    public static void texture(DepthTexture texture, Float2 size) {
        GUIDisplay.texture(texture.getBuffer(), size.x, size.y);
    }

    public static void texture(int texture, float width, float height) {
        ImGui.image(texture, width, height, 0, 1, 1, 0);
    }
}

File: source/gui/helper/GUIDisplay.kt
package gui.helper

import glparts.DepthTexture
import glparts.Texture
import imgui.ImGui
import math.Float2
import math.Int2

object GUIDisplay {
    fun texture(texture: Texture?, size: Int2?) {
        texture(texture, Float2(size))
    }

    fun texture(texture: DepthTexture?, size: Int2?) {
        texture(texture, Float2(size))
    }

    fun texture(texture: Texture?, size: Float2) {
        texture(texture!!.buffer, size.x, size.y)
    }

    fun texture(texture: DepthTexture?, size: Float2) {
        texture(texture!!.buffer, size.x, size.y)
    }

    fun texture(texture: Int, width: Float, height: Float) {
        ImGui.image(texture, width, height, 0f, 1f, 1f, 0f)
    }
}